### DISCUSSION TITLE
Celery requirements.txt files should PEP 440

### DISCUSSION THREAD

**Discussion by norg** [celery/celery#8001] [Q&A] [answered]
Hi,

while running `pip-audit` I ran into an issue that was revealed by `packaging` which brought me to `celery` in the end.
The following output was seen:
```
packaging.requirements.InvalidRequirement: Expected closing RIGHT_PARENTHESIS
    pytz (>dev)
```
I found that this was set here for example: https://github.com/celery/celery/blob/main/requirements/dev.txt

This doesn't follow https://peps.python.org/pep-0440/ and thus the error is valid IMHO.

In the past tools like `packaging` haven't been that strict but now are. So I would recommend to update the requirements.txt file to follow the PEP440 standard unless there are arguments for that syntax that I'm not aware of.

I did not create an issue yet, since the issue template recommended to use the discussions forum first.

THanks

**Comment by auvipy** [comment]
will this fix this? https://github.com/celery/celery/pull/8002

**Reply by pradyunsg** [reply]
No.

You can try passing these to `packaging` (specifically, `packaging.requirements.Requirement`) to see what is acceptable.

**Reply by auvipy** [reply]
OK will crosscheck

**Reply by norg** [reply]
Is there a reason why `>dev` is used? Wouldn't it make more sense to use `>2022.4` for example? This would reflect the versioning from `pytz`.
I played a round with the requirements from `packaging` and as soon as there is a number in front, parsing is happy.
Although I must admit that the error message from `packaging` is a bit misleading with pointing to `RIGHT_PARENTHESIS`.

**Reply by auvipy** [reply]
i changed to numbered version

**Reply by pradyunsg** [reply]
That error is better with https://github.com/pypa/packaging/pull/662

**Comment by edmorley** [comment, marked as answer]
(Cross-linking a few issues, since the pip 24.1+ changes are going to break quite a few older apps)

This was fixed in Celery 5.2.1+ too - see:
https://github.com/celery/celery/pull/7074
